dCommon/SHA512.cpp

```diff
@@ -51,7 +51,7 @@ void SHA512::transform(const unsigned char* message, unsigned int block_nb) {
 	uint64 t1, t2;
 	const unsigned char* sub_block;
 	int i, j;
-	for (i = 0; i < (int)block_nb; i++) {
+	for (i = 0; i < static_cast<int>(block_nb); i++) {
```

Comment:
the changes in this file should be undone

Reply:
Are we certain it is vendored? The closest I can find from a brief search is [this one,](https://www.geeksforgeeks.org/implement-secure-hashing-algorithm-512-sha-512-as-functional-programming-paradigm/) and even that's quite different in implementation.